### PR TITLE
gameOfLife

### PR DESCRIPTION
**Pull request by SheyReiff**

### CODE REVIEW

File: src/main/java/reiff/gameoflife/GameOfLife.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        int[][] nextField = new int[height][width];
+
+
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                int liveNeighbors = countLiveNeighbors(x, y);
+
+                if (field[y][x] == 1) {
+                    if (liveNeighbors < 2 || liveNeighbors > 3) {
+                        nextField[y][x] = 0;
```
Comment (andrewoid) [inline src/main/java/reiff/gameoflife/GameOfLife.java]: No need to do this. It's already 0

File: src/main/java/reiff/gameoflife/GameOfLife.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    } else {
+                        nextField[y][x] = 1;
+                    }
+                } else {
+                    if (liveNeighbors == 3) {
+                        nextField[y][x] = 1;
+                    } else {
+                        nextField[y][x] = 0;
```
Comment (andrewoid) [inline src/main/java/reiff/gameoflife/GameOfLife.java]: Same as above

File: src/main/java/reiff/gameoflife/GameOfLife.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                if (neighborOffsetX == 0 && neighborOffsetY == 0) {
+                    continue;
+                }
+
+                int neighborX = x + neighborOffsetX;
+                int neighborY = y + neighborOffsetY;
+
+
+                if (neighborX >= 0 && neighborX < width && neighborY >= 0 && neighborY < height) {
+                    liveCount += field[neighborY][neighborX];
```
Comment (andrewoid) [inline src/main/java/reiff/gameoflife/GameOfLife.java]: LiveCount++

File: src/test/java/reiff/gameoflife/GameOfLifeTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+
+        @Test
+    public void nextGen() {
+        //given
+        GameOfLife game = new GameOfLife(3, 3);
+        game.setCell(1, 1, 1);
+        game.setCell(2, 1, 1);
+        game.setCell(0, 0, 1);
```
Comment (andrewoid) [inline src/test/java/reiff/gameoflife/GameOfLifeTest.java]: 3 horizontal or 3 vertical cells

Comment (SheyReiff) [reply src/test/java/reiff/gameoflife/GameOfLifeTest.java]: I should do either "000\n111\n000\n" or "010\n010\n010\n"?

Comment (andrewoid) [reply src/test/java/reiff/gameoflife/GameOfLifeTest.java]: Yes

Comment (SheyReiff) [reply src/test/java/reiff/gameoflife/GameOfLifeTest.java]: Thank you

**Comment by andrewoid** [comment]
Seems like there's a failing test.

File: src/main/java/reiff/gameoflife/GameOfLife.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                if (offsetX == 0 && offsetY == 0) {
+                    continue;  // Skip the cell itself
+                }
+
+                int neighborX = x + offsetX;
+                int neighborY = y + offsetY;
+
+
+                if (neighborX >= 0 && neighborX < width && neighborY >= 0 && neighborY < height) {
+                    if (field[neighborY][neighborX] == 1) {
```
Comment (andrewoid) [inline src/main/java/reiff/gameoflife/GameOfLife.java:64]: You can combine these

**Review by andrewoid — APPROVED** [review]

**Pull request merged**